system/jlib/jstats.cpp

```diff
@@ -400,40 +399,53 @@ static StringBuffer & formatIPV4(StringBuffer & out, unsigned __int64 value)
     return out.appendf("%d.%d.%d.%d", ip1, ip2, ip3, ip4);
 }
 
-static const char *getMoneyLocale()
+class MoneyLocale
 {
-    static bool haveLocale = false;
-    static StringAttr moneyLocale;
-
-    if (!haveLocale)
+public:
+    ~MoneyLocale()
+    {
+        std::locale * tmp = nullptr;
+        locale.exchange(tmp, std::memory_order_relaxed);
+        delete tmp;
```

Comment:
minor: "delete locale.load()" is sufficient - this object is being destroyed, so it can't be accessed from any other thread